scipy/sparse/_construct.py

```diff
@@ -70,39 +79,32 @@ def spdiags(data, diags, m=None, n=None, format=None):
     return dia_matrix((data, diags), shape=(m, n)).asformat(format)
 
 
-def diags(diagonals, offsets=0, shape=None, format=None, dtype=None):
+def diags_array(diagonals, offsets=0, *, shape=None, format=None, dtype=None):
```

Comment:
While we're here, let's also make `diagonals` positional-only using the `/` feature.